ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -246,11 +250,32 @@ public void handlePost(RestRequest restRequest, RestResponseChannel restResponse
 
   @Override
   public void handlePut(RestRequest restRequest, RestResponseChannel restResponseChannel) {
+    long processingStartTime = System.currentTimeMillis();
+    long preProcessingTime = 0;
     handlePrechecks(restRequest, restResponseChannel);
-    Exception exception =
-        isUp ? new RestServiceException("PUT is not supported", RestServiceErrorCode.UnsupportedHttpMethod)
-            : new RestServiceException("AmbryBlobStorageService unavailable", RestServiceErrorCode.ServiceUnavailable);
-    submitResponse(restRequest, restResponseChannel, null, exception);
+    try {
+      logger.trace("Handling PUT request - {}", restRequest.getUri());
+      checkAvailable();
+      // TODO: make this non blocking once all handling of individual methods is moved to their own classes
+      securityService.preProcessRequest(restRequest).get();
```

Comment:
should this call happen inside of ttlUpdateHandler?

Reply:
I'm not sure - I saw that you did this in the `PostBlobHandler` but it seems like the preprocessing is actually not related to the kind of request - I'm thinking that we can unify the code as one function is `AmbryBlobStorageService` - what do you think?